x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -57,6 +57,7 @@ async function createSetupSideEffects(
     ensureInstalledDefaultPackages(soClient, callCluster),
     outputService.ensureDefaultOutput(soClient),
     agentPolicyService.ensureDefaultAgentPolicy(soClient),
+    setupFleet(soClient, callCluster),
```

Comment:
This will setup Agents if the user do not explicitly enable Agents, not sure it's what we want

Reply:
@nchaulet yea, I'm seeing odd behavior right now.

I'm thinking we do something like this:

Take the call to create the role here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/server/services/setup.ts#L133

And break it out to its own function, something like `updateFleetRole()` where we can isolate that call and check for the existence first.

Reply:
Yes I think what you propose make sense